src/Common/tests/System/AssertExtensions.cs

```diff
@@ -50,5 +50,26 @@ public static T Throws<T>(string paramName, Func<object> testCode)
 
             return exception;
         }
+
+        public static void Throws(Type netCoreExceptionType, string netCoreParamName, Type netFxExceptionType, string netFxParamName, Action action)
+        {
+            ArgumentException exception;
+
+            if (RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework"))
+            {
+                exception = (ArgumentException)Assert.Throws(netFxExceptionType, action);
+            }
+            else
+            {
+                exception = (ArgumentException)Assert.Throws(netCoreExceptionType, action);
+            }
```

Comment:
If you don't change the signature above, this can be consolidated to just:
```C#
var exception = (ArgumentException)Assert.Throws(
    RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework") ? netFxExceptionType : netCoreExceptionType,
    action);
```